lint-staged.config.js

```diff
@@ -0,0 +1,19 @@
+//eslint-disable-next-line @typescript-eslint/no-var-requires
+const { ESLint } = require('eslint');
+
+const removeIgnoredFiles = async (files) => {
+	const eslint = new ESLint();
+	const ignoredFiles = await Promise.all(
+		files.map((file) => eslint.isPathIgnored(file))
+	);
+	const filteredFiles = files.filter((_, i) => !ignoredFiles[i]);
+	return filteredFiles.join(' ');
+};
+
+module.exports = {
```

Comment:
I had to move lint-staged config from `.lintstagedrc` to `lint-staged.config.js` so files from `.eslintignore` can be ignored properly. [This is the official way to do so from lint-staged docs](https://github.com/okonet/lint-staged#how-can-i-ignore-files-from-eslintignore).